src/Foundation/NSUrl.cs

```diff
@@ -30,6 +30,32 @@ public NSUrl (string path, string relativeToUrl)
 		{
 		}
 
+		public NSUrl (string urlString)
+		{
+			Handle = _Init (urlString);
+			if (Handle == IntPtr.Zero)
+				throw new ArgumentException ("Malformed url");
+		}
+
+		public NSUrl (string urlString, NSUrl relativeToUrl)
+		{
+			Handle = _Init (urlString, relativeToUrl);
+			if (Handle == IntPtr.Zero)
+				throw new ArgumentException ("Malformed url");
+		}
+
+		// Call these instead of obj-c static versions since these throw
+		// if nil returned due to malformed input
```

Comment:
Having a better/different exception on `initWithString:` is fine.

OTOH chasing the behaviour of `FromString` would be really bad (at least for iOS where existing code expect `null` for invalid URL)